src/resources/general/compiler-development.md

```diff
@@ -0,0 +1,43 @@
+# 🛠️ Compiler Development
+
+This is a collection of resources on compiler construction, programming language design, and related topics.
+
+## 📚 Compiler Resources
+
+- [_Engineering a Compiler_](https://www.elsevier.com/books/engineering-a-compiler/cooper/978-0-12-815412-0)
+  <sub>Great for compiler back-ends and optimization.</sub>
+
+- [_Crafting Interpreters_](https://craftinginterpreters.com/)
```

Comment:
I think we want these to all be part of the same list without a paragraph break between